google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java

```diff
@@ -911,11 +935,14 @@ static final class AppendRequestAndResponse {
     // The writer that issues the call of the request.
     final StreamWriter streamWriter;
 
+    final Instant requestCreationTimeStamp;
+
     AppendRequestAndResponse(AppendRowsRequest message, StreamWriter streamWriter) {
       this.appendResult = SettableApiFuture.create();
       this.message = message;
       this.messageSize = message.getProtoRows().getSerializedSize();
       this.streamWriter = streamWriter;
+      this.requestCreationTimeStamp = Instant.now();
```

Comment:
Not sure if this is 100% right, since this gets called when the request is added to waitingRequestQueue. I think we want a timestamp that is set  close to when the actual RPC goes out.

Reply:
I was trying to start counting time when user call Append. But let's change to set the timestamp when the request is added to the inflight queue.

Reply:
The idea is to detect stuck RPCs. If things are stuck in the waitingRequestQueue (and nothing is stuck in the inflight queue), then presumably something else is very wrong.